### PR TITLE
fix wrong `Label.frame_body.x2` value

### PR DESCRIPTION
**Pull request by TymurGubayev**
`update_scroll_inset` might change `frame_inset`, i.e. we need to `computeFrame` with the new values.

### CODE REVIEW

**Comment by TymurGubayev** [comment]
> How hard would it be to write a unit test for this?

I think it's possible, I'll try to write something in the next few days.

**Comment by TymurGubayev** [comment]
added a few tests.

File: test/library/gui/widgets.Label.lua [hunk 1]
```diff
@@ -0,0 +1,117 @@
+-- test -dhack/scripts/devel/tests -twidgets.Label
+
+--reload('gui')
+--reload('gui.widgets')
```
Comment (myk002) [inline test/library/gui/widgets.Label.lua:4]: this should not be necessary. the test harness loads all lua libraries cleanly before starting the tests.

File: test/library/gui/widgets.Label.lua [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    fs.ATTRS = {
+        frame_style = gui.GREY_LINE_FRAME,
+        frame_title = 'TestFramedScreen',
+        frame_width = 10,
+        frame_height = 10,
+        frame_inset = 0,
+        focus_path = 'test-framed-screen',
+    }
+
+    t = {}
```
Comment (myk002) [inline test/library/gui/widgets.Label.lua:25]: missing `local`? Also in the other tests.

File: test/library/gui/widgets.Label.lua [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+function test.Label_correct_frame_body_with_scroll_icons()
+    local fs = defclass(fs, gui.FramedScreen)
+    fs.ATTRS = {
+        frame_style = gui.GREY_LINE_FRAME,
+        frame_title = 'TestFramedScreen',
+        frame_width = 10,
+        frame_height = 10,
+        frame_inset = 0,
+        focus_path = 'test-framed-screen',
+    }
```
Comment (myk002) [inline test/library/gui/widgets.Label.lua:23]: You can declare this class outside of the function so you can reuse it across tests.

File: test/library/gui/widgets.Label.lua [hunk 4]
```diff
@@ -0,0 +1,117 @@
+-- test -dhack/scripts/devel/tests -twidgets.Label
```
Comment (myk002) [inline test/library/gui/widgets.Label.lua:1]: There is actually a `test/library/gui/widgets.lua` file that you can add these tests to now. It was created relatively recently, so you might have to merge latest develop to see it.

**Review by myk002 — CHANGES_REQUESTED** [review]
The tests are good to have, but I notice that they don't actually test the change being made here -- the tests pass just fine if the change to `widgets.lua` is reverted.

**Comment by TymurGubayev** [comment]
> The tests are good to have, but I notice that they don't actually test the change being made here -- the tests pass just fine if the change to `widgets.lua` is reverted.

huh, that's strange: the first one (`Label_correct_frame_body_with_scroll_icons`) fails for me without the changes.

File: test/library/gui/widgets.Label.lua (hunk 4 above)
Comment (TymurGubayev) [reply test/library/gui/widgets.Label.lua:1]: yep, wasn't there last time I checked. But maybe separate widget test files aren't bad? I imagine should every widget get tests, the single file with all the tests will be huge.

**Comment by myk002** [comment]
> > The tests are good to have, but I notice that they don't actually test the change being made here -- the tests pass just fine if the change to `widgets.lua` is reverted.
> 
> huh, that's strange: the first one (`Label_correct_frame_body_with_scroll_icons`) fails for me without the changes.

I must have been in an inconsistent state. I'll check through again on Monday. Sorry!

File: test/library/gui/widgets.Label.lua (hunk 4 above)
Comment (lethosor) [reply test/library/gui/widgets.Label.lua:1]: I'm ok with separate per-widget tests

**Comment by lethosor** [comment]
I can confirm that one of the three tests fails for me with the implementation taken from `develop` (the other two tests pass with either implementation):

```
Check failed! Label's frame_body.x2 and .width should be one smaller because of show_scroll_icons.: "10" ~= "9"
  at .../test/library/gui/widgets.Label.lua:42

test failed: library/gui/widgets.Label:Label_correct_frame_body_with_scroll_icons
```

I did forget to re-run `ninja install` at first - maybe double-check that you have the right files installed in `hack/lua`?

**Review by myk002 — APPROVED** [review]

**Pull request merged**